include/config/dexnav.h

```diff
@@ -0,0 +1,72 @@
+#ifndef GUARD_CONFIG_DEXNAV_H
+#define GUARD_CONFIG_DEXNAV_H
+
+#define DEXNAV_ENABLED                FALSE  // Whether or not dexnav is enabled. If TRUE, flags/vars below must all be non-zero
+#define USE_DEXNAV_SEARCH_LEVELS      FALSE  /* WARNING: POSSIBLY EXCEEDS SAVEBLOCK SPACE! REQUIRES 1 BYTE PER SPECIES */
+
+// Flag/var defines
+#define FLAG_SYS_DEXNAV_SEARCH          0 // Searching for mon
+#define FLAG_SYS_DEXNAV_GET             0 // Dexnav shows in start menu
+#define FLAG_SYS_DETECTOR_MODE          0 // Allow player to find hidden mons
+#define VAR_DEXNAV_SPECIES              0 // Registered dexnav species
+#define VAR_DEXNAV_STEP_COUNTER         0 // Steps for finding hidden pokemon
```

Comment:
Final request before we merge this in: these five are all needed if DexNav is enabled, which means we should probably add a static assert to make sure these are set before people complain about UB related to exactly this. Thoughts?